src/lib/server/services/content.ts

```diff
@@ -268,6 +272,70 @@ export const getPostStatistics = async (db: DB) => {
 };
 export type PostStatistics = Awaited<ReturnType<typeof getPostStatistics>>;
 
+export const getCommentTree = async (db: DB, post_id: string): Promise<CommentData> => {
+    const firstLevelComments = await db
+        .select()
+        .from(postTable)
+        .where(and(eq(postTable.id, post_id), isNull(commentTable.replyTo)))
+        .innerJoin(commentTable, eq(commentTable.postId, postTable.id))
+        .orderBy(commentTable.createdOn)
+        .limit(PAGE_SIZE);
+
+    console.log(
+        firstLevelComments.length +
+            ' top-level comment' +
+            (firstLevelComments.length != 1 ? 's' : '') +
+            ' on this post.'
+    );
+    const firstLevelCommentsIds = firstLevelComments.map((p) => p.comment.id);
+
+    const secondLevelComments = await db
+        .select()
+        .from(postTable)
+        .where(and(eq(postTable.id, post_id), inArray(commentTable.replyTo, firstLevelCommentsIds)))
+        .innerJoin(commentTable, eq(commentTable.postId, postTable.id))
+        .orderBy(commentTable.createdOn)
+        .limit(PAGE_SIZE / 2);
```

Comment:
Not sure how limit will work with non-integer values, should add a `Math.floor` here.